File: .archive/legacy-scripts/create-app-icon.js
const sharp = require('sharp');
const path = require('path');

async function createAppIcon() {
  const assetsDir = path.join(__dirname, 'assets');
  const logoPath = path.join(assetsDir, 'logo.png');
  const iconPath = path.join(assetsDir, 'icon.png');
  const adaptiveIconPath = path.join(assetsDir, 'adaptive-icon.png');

  const size = 1024;

  // Since the logo already has a black background with white whale and red eye,
  // we just need to resize it to the proper icon size
  await sharp(logoPath)
    .resize(size, size, {
      fit: 'contain',
      background: { r: 0, g: 0, b: 0, alpha: 1 }
    })
    .png()
    .toFile(iconPath);

  // Copy to adaptive icon
  await sharp(iconPath).toFile(adaptiveIconPath);

  console.log('✅ Created app icons with whale logo:');
  console.log(`  - ${iconPath}`);
  console.log(`  - ${adaptiveIconPath}`);
}

createAppIcon().catch(console.error);

File: .archive/create-app-icon.py
#!/usr/bin/env python3
"""
Create app icon with black background
"""
from PIL import Image
import os

# Paths
assets_dir = os.path.dirname(os.path.abspath(__file__)) + '/assets'
logo_path = os.path.join(assets_dir, 'logo.png')
icon_path = os.path.join(assets_dir, 'icon.png')
adaptive_icon_path = os.path.join(assets_dir, 'adaptive-icon.png')

# Open the logo
logo = Image.open(logo_path)

# Create 1024x1024 black background
size = 1024
icon = Image.new('RGB', (size, size), (0, 0, 0))

# Resize logo to fit nicely (60% of icon size)
logo_size = int(size * 0.6)
logo_resized = logo.resize((logo_size, logo_size), Image.Resampling.LANCZOS)

# Calculate position to center the logo
position = ((size - logo_size) // 2, (size - logo_size) // 2)

# Paste logo onto black background
# If logo has transparency, use it as mask
if logo_resized.mode in ('RGBA', 'LA'):
    icon.paste(logo_resized, position, logo_resized)
else:
    icon.paste(logo_resized, position)

# Save as icon.png and adaptive-icon.png
icon.save(icon_path, 'PNG')
icon.save(adaptive_icon_path, 'PNG')

print(f"✅ Created app icons with black background:")
print(f"  - {icon_path}")
print(f"  - {adaptive_icon_path}")
